### ISSUE TITLE
libv8-node install issues

### ISSUE THREAD

**Issue by garethrees** [mysociety/alaveteli#6229]
On 2021-05-04 I updated develop and started getting a weird problem with `libv8-node`, which is a new dependency since 55a0460c7fbd96bcaab69a12b31d6f6f863d0c85.

```
$ bin/rspec spec/models/info_request_spec.rb
Could not find libv8-node-15.14.0.1 in any of the sources
Run `bundle install` to install missing gems.
```
Run bundle install again and get the same thing. It seems to keep trying to fetch/install that gem each time I run bundle install:

```
Fetching libv8-node 15.14.0.1 (x86_64-linux-musl)
Installing libv8-node 15.14.0.1 (x86_64-linux-musl)
```

On a fresh bionic VM, I'm getting:

```
vagrant@10:~/alaveteli$ bundle exec rake -T
rake aborted!
LoadError: /home/vagrant/bundle/ruby/2.5.0/extensions/x86_64-linux/2.5.0/mini_racer-0.4.0/mini_racer_extension.so: undefined symbol: _ZNSt7__cxx1119basic_ostringstreamIcSt11char_traitsIcESaIcEEC1Ev
/home/vagrant/bundle/ruby/2.5.0/gems/mini_racer-0.4.0/lib/mini_racer.rb:11:in `load'
/home/vagrant/bundle/ruby/2.5.0/gems/mini_racer-0.4.0/lib/mini_racer.rb:11:in `<top (required)>'
/home/vagrant/alaveteli/config/application.rb:21:in `<top (required)>'
/home/vagrant/alaveteli/rakefile:4:in `require_relative'
/home/vagrant/alaveteli/rakefile:4:in `<top (required)>'
/home/vagrant/bundle/ruby/2.5.0/gems/rake-13.0.3/exe/rake:27:in `<top (required)>'
(See full trace by running task with --trace)
```

And then after rebuilding my "normal" VM (`vagrant destroy -f && vagrant up`) I get:

```
    default: Fetching zip_tricks 5.5.0
    default: Installing zip_tricks 5.5.0
    default: Bundle complete! 86 Gemfile dependencies, 168 gems now installed.
    default: Bundled gems are installed into `/home/vagrant/bundle`
    default: Post-install message from gender_detector:
    default: For unicode support you'll need to also install the unicode_utils or activesupport gem
    default: Post-install message from icalendar:
    default: ActiveSupport is required for TimeWithZone support, but not required for general use.
    default: /usr/lib/ruby/vendor_ruby/bundler/spec_set.rb:88:in `block in materialize'
    default: :
    default: Could not find libv8-node-15.14.0.1 in any of the sources
    default:  (
    default: Bundler::GemNotFound
    default: )
    default:    from /usr/lib/ruby/vendor_ruby/bundler/spec_set.rb:82:in `map!'
    default:    from /usr/lib/ruby/vendor_ruby/bundler/spec_set.rb:82:in `materialize'
    default:    from /usr/lib/ruby/vendor_ruby/bundler/definition.rb:170:in `specs'
    default:    from /usr/lib/ruby/vendor_ruby/bundler/definition.rb:237:in `specs_for'
    default:    from /usr/lib/ruby/vendor_ruby/bundler/definition.rb:226:in `requested_specs'
    default:    from /usr/lib/ruby/vendor_ruby/bundler/runtime.rb:108:in `block in definition_method'
    default:    from /usr/lib/ruby/vendor_ruby/bundler/runtime.rb:20:in `setup'
    default:    from /usr/lib/ruby/vendor_ruby/bundler.rb:107:in `setup'
    default:    from /usr/lib/ruby/vendor_ruby/bundler/setup.rb:20:in `<top (required)>'
    default:    from /usr/lib/ruby/2.5.0/rubygems/core_ext/kernel_require.rb:59:in `require'
    default:    from /usr/lib/ruby/2.5.0/rubygems/core_ext/kernel_require.rb:59:in `require'
```

_EDIT:_ the above is through `vagrant provision`. On running `bundle` manually in an ssh session, I get the same problem of it repeatedly trying to install the gem.

**Comment by garethrees** [comment]
```
$ bundle pristine
Could not find libv8-node-15.14.0.1 in any of the sources

$ bundle pristine mini_racer
Could not find libv8-node-15.14.0.1 in any of the sources
```

**Comment by garethrees** [comment]
This is strange, in that it looks like there's only an issue with the _bundled_ version…

```
vagrant@10:~/alaveteli$ ruby -e 'require "mini_racer"; puts "OK"'
OK

vagrant@10:~/alaveteli$ bundle exec ruby -e 'require "mini_racer"; puts "OK"'
Loading..... not find mini_racer_extension.so in ["/home/vagrant/bundle/ruby/2.5.0/extensions/x86_64-linux/2.5.0/mini_racer-0.4.0", "/home/vagrant/bundle/ruby/2.5.0/gems/mini_racer-0.4.0/lib", "/home/vagrant/bundle/ruby/2.5.0/gems/mini_racer-0.4.0/ext"]: /home/vagrant/bundle/ruby/2.5.0/extensions/x86_64-linux/2.5.0/mini_racer-0.4.0/mini_racer_extension.so
Traceback (most recent call last):
        3: from -e:1:in `<main>'
        2: from -e:1:in `require'
        1: from /home/vagrant/bundle/ruby/2.5.0/gems/mini_racer-0.4.0/lib/mini_racer.rb:12:in `<top (required)>'
/home/vagrant/bundle/ruby/2.5.0/gems/mini_racer-0.4.0/lib/mini_racer.rb:12:in `load': /home/vagrant/bundle/ruby/2.5.0/extensions/x86_64-linux/2.5.0/mini_racer-0.4.0/mini_racer_extension.so: undefined symbol: _ZNSt7__cxx1119basic_ostringstreamIcSt11char_traitsIcESaIcEEC1Ev (LoadError)

```

**Comment by gbp** [comment]
Using that Vagrantfile, I see:

```
==> default: Running provisioner: shell...
    default: Running: inline script
    default: Fetching gem metadata from https://rubygems.org/
    default: .
    default: .
    default: Resolving dependencies...
    default: Using bundler 2.2.17
    default: Fetching libv8-node 15.14.0.1 (x86_64-linux-musl)
    default: Fetching libv8-node 15.14.0.1 (x86_64-linux)
    default: Installing libv8-node 15.14.0.1 (x86_64-linux-musl)
    default: Installing libv8-node 15.14.0.1 (x86_64-linux)
    default: Fetching mini_racer 0.4.0
    default: Installing mini_racer 0.4.0 with native extensions
    default: Gem::Ext::BuildError: ERROR: Failed to build gem native extension.
    default:
    default: current directory:
    default: /vagrant/vendor/bundle/ruby/2.5.0/gems/mini_racer-0.4.0/ext/mini_racer_loader
    default: /usr/bin/ruby2.5 -r ./siteconf20210512-3122-5x3mtx.rb extconf.rb
    default: creating Makefile
    default:
    default: current directory:
    default: /vagrant/vendor/bundle/ruby/2.5.0/gems/mini_racer-0.4.0/ext/mini_racer_loader
    default: make "DESTDIR=" clean
    default: sh: 1: make: not found
    default:
    default: current directory:
    default: /vagrant/vendor/bundle/ruby/2.5.0/gems/mini_racer-0.4.0/ext/mini_racer_loader
    default: make "DESTDIR="
    default: sh: 1: make: not found
    default:
    default: make failed, exit code 127
    default:
    default: Gem files will remain installed in
    default: /vagrant/vendor/bundle/ruby/2.5.0/gems/mini_racer-0.4.0 for inspection.
    default: Results logged to
    default: /vagrant/vendor/bundle/ruby/2.5.0/extensions/x86_64-linux/2.5.0/mini_racer-0.4.0/gem_make.out
    default:
    default: An error occurred while installing mini_racer (0.4.0), and Bundler cannot
    default: continue.
    default: Make sure that `gem install mini_racer -v '0.4.0' --source
    default: 'https://rubygems.org/'` succeeds before bundling.
```

**Comment by gbp** [comment]
With `sudo apt-get install build-essential`, I see the same as https://github.com/mysociety/alaveteli/issues/6229#issuecomment-838510874

**Comment by garethrees** [comment]
Given the notes about a more recent GCC in https://github.com/rubyjs/mini_racer/issues/203, this might get solved by #6294 / #5637.

**Comment by garethrees** [comment]
This got fixed (for me) by #6392.
